lib/modules/webserver/server.js

```diff
@@ -31,15 +33,17 @@ class Server {
     const main = serveStatic(this.buildDir, {'index': ['index.html', 'index.htm']});
 
     this.app = express();
+    expressWebSocket(this.app);
+
     this.app.use(main);
     this.app.use('/coverage', coverage);
     this.app.use(coverageStyle);
 
-    expressWebSocket(this.app);
-
     this.app.ws('/', function(ws, _req) {
       self.events.on('outputDone', () => {
-        ws.send('outputDone');
+        if (ws.readyState === WebSocketOpen) {
```

Comment:
How about going with `!!ws.readyState` instead here?

Reply:
And then remove `WebSocketOpen`

Reply:
What if `ws.readyState` is equal to 2 or 3
https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/readyState

I guess we really want to check for the exact value 1
What do you think?

Reply:
Ah, good point! In that case, I wonder if it would be nicer to have those states represented as well.

```
const WebSocktState = {
  Open: 1,
  ...
}
```

Enums are perfect for that but those we only get with TypeScript.

Reply:
On a second thought, I'd say this can just stay as is and may be adjusted as needed. The only nitpick I'd add is to change `WebSocketOpen` to `WEB_SOCKET_STATE_OPEN`.

Reply:
Agree with the second thought.